[CI] AddIndexBlockRollingUpgradeIT testAddBlock {upgradedNodes=2} failing

**Build Scans:**
- [elasticsearch-intake #16936 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/pj6rjalerhs6y)
- [elasticsearch-intake #16934 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2rlryjp76rosk)
- [elasticsearch-intake #16932 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mkbb5lcdfj3r6)

**Reproduction Line:**
```
./gradlew ":qa:rolling-upgrade:v8.19.0#bwcTest" -Dtests.class="org.elasticsearch.upgrades.AddIndexBlockRollingUpgradeIT" -Dtests.method="testAddBlock {upgradedNodes=1}" -Dtests.seed=9DE645285488F16E -Dtests.bwc=true -Dtests.locale=jgo -Dtests.timezone=Asia/Manila -Druntime.java=23
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.AddIndexBlockRollingUpgradeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAddBlock%20%7BupgradedNodes%3D2%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.RuntimeException: An error occurred while checking cluster 'test-cluster' status.
```

**Issue Reasons:**
- [9.0] 3 consecutive failures in step 8.19.0_bwc-snapshots
- [9.0] 3 failures in test testAddBlock {upgradedNodes=2} (5.8% fail rate in 52 executions)
- [9.0] 3 failures in step 8.19.0_bwc-snapshots (18.8% fail rate in 16 executions)
- [9.0] 3 failures in pipeline elasticsearch-intake (25.0% fail rate in 12 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

The BWC upgrade test failed to upgrade the cluster from 8.19.0-SNAPSHOT to 9.0.0-SNAPSHOT because the 9.0 node is lacking a required feature:

> [test-cluster-1][127.0.0.1:37265][internal:cluster/coordination/join]","error.stack_trace":"org.elasticsearch.transport.RemoteTransportException: [test-
cluster-1][127.0.0.1:37265][internal:cluster/coordination/join]\nCaused by: java.lang.IllegalStateException: Node AOrAY7ayQAmKRlqmV4SunA is missing required features [mapper.copy_to.dynamic_
handling]

This feature was introduced few hours ago by #121367 which was merged on 9.1 and 8.x (8.19) branches but not in the 9.0 branch. I asked the relevant team about this issue, and I'm keeping this issue open to not forget about unmuting the test later.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -334,8 +334,6 @@ tests:
 - class: org.elasticsearch.upgrades.VectorSearchIT
   method: testBBQVectorSearch {upgradedNodes=0}
   issue: https://github.com/elastic/elasticsearch/issues/121253
-- class: org.elasticsearch.lucene.FullClusterRestartLuceneIndexCompatibilityIT
-  issue: https://github.com/elastic/elasticsearch/issues/121257
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/121269
 - class: org.elasticsearch.upgrades.VectorSearchIT
@@ -366,23 +364,12 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSuggestProfilesWithHint
   issue: https://github.com/elastic/elasticsearch/issues/121116
-- class: org.elasticsearch.upgrades.AddIndexBlockRollingUpgradeIT
-  method: testAddBlock {upgradedNodes=2}
-  issue: https://github.com/elastic/elasticsearch/issues/121365
-- class: org.elasticsearch.upgrades.AddIndexBlockRollingUpgradeIT
-  method: testAddBlock {upgradedNodes=3}
-  issue: https://github.com/elastic/elasticsearch/issues/121366
-- class: org.elasticsearch.upgrades.AddIndexBlockRollingUpgradeIT
-  method: testAddBlock {upgradedNodes=1}
-  issue: https://github.com/elastic/elasticsearch/issues/121367
 - class: org.elasticsearch.env.NodeEnvironmentTests
   method: testGetBestDowngradeVersion
   issue: https://github.com/elastic/elasticsearch/issues/121316
 - class: org.elasticsearch.ingest.geoip.FullClusterRestartIT
   method: testGeoIpSystemFeaturesMigration {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/121115
-- class: org.elasticsearch.lucene.RollingUpgradeLuceneIndexCompatibilityTestCase
-  issue: https://github.com/elastic/elasticsearch/issues/121423
 - class: org.elasticsearch.index.engine.ShuffleForcedMergePolicyTests
   method: testDiagnostics
   issue: https://github.com/elastic/elasticsearch/issues/121336

--- a/qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/FullClusterRestartLuceneIndexCompatibilityIT.java
+++ b/qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/FullClusterRestartLuceneIndexCompatibilityIT.java
@@ -11,8 +11,6 @@ package org.elasticsearch.lucene;
 
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.index.IndexSettings;
-import org.elasticsearch.index.translog.Translog;
 import org.elasticsearch.repositories.fs.FsRepository;
 import org.elasticsearch.test.cluster.util.Version;
 
@@ -184,7 +182,6 @@ public class FullClusterRestartLuceneIndexCompatibilityIT extends FullClusterRes
                 Settings.builder()
                     .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
                     .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, randomInt(2))
-                    .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), randomFrom(Translog.Durability.values()))
                     .build()
             );
             indexDocs(index, numDocs);

--- a/qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/RollingUpgradeLuceneIndexCompatibilityTestCase.java
+++ b/qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/RollingUpgradeLuceneIndexCompatibilityTestCase.java
@@ -13,8 +13,6 @@ import org.elasticsearch.client.Request;
 import org.elasticsearch.client.ResponseException;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.index.IndexSettings;
-import org.elasticsearch.index.translog.Translog;
 import org.elasticsearch.repositories.fs.FsRepository;
 import org.elasticsearch.test.cluster.util.Version;
 
@@ -189,11 +187,7 @@ public class RollingUpgradeLuceneIndexCompatibilityTestCase extends RollingUpgra
             createIndex(
                 client(),
                 index,
-                Settings.builder()
-                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
-                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
-                    .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), randomFrom(Translog.Durability.values()))
-                    .build()
+                Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build()
             );
             indexDocs(index, numDocs);
             return;

